pennylane/ops/qubit/matrix_ops.py

```diff
@@ -236,10 +247,17 @@ def has_decomposition(self) -> bool:
 
     def adjoint(self) -> "QubitUnitary":
         U = self.matrix()
+        if isinstance(U, csr_matrix):
+            adjoint_sp_mat = U.conjugate().transpose()
+            # Note: it is necessary to explicitly cast back to csr, or it will be come csc
```

Comment:
```suggestion
            # Note: it is necessary to cast back to csr explicitly, or it will become csc
```

Curious, why does this happen?

Reply:
@PietropaoloFrisoni SciPy has its own optimization over matrix manipulation and storage. Whenever a CSR got transposed it automatically convert the obj to CSC no matter the actual matrix structure.
```python
m1 = sparse.csr_matrix([[1, 0], [0, 1]])
print(m1) # CSR
m2 = m1.T
print(m2) # CSC
```